File: 0573-squirrel-simulation/0573-squirrel-simulation.py
class Solution:
    def minDistance(self, height: int, width: int, tree: List[int], squirrel: List[int], nuts: List[List[int]]) -> int:
        def distance(x: int, y: int, target: List[int]):
            return abs(x - target[0]) + abs(y - target[1])
        
        total_dist, squirrel_to_closest = 0, float('inf')
        for x, y in nuts:
            total_dist +=  2 * distance(x, y, tree)
            # Find the closest distance in which a nut is located from a squirrel, considering to deduct one-way distance between a nut and tree
            squirrel_to_closest = min(squirrel_to_closest, -distance(x, y, tree) + distance(x, y, squirrel))
            
        return total_dist + squirrel_to_closest

File: 0573-squirrel-simulation/0573-squirrel-simulation.cpp
class Solution {
private:
    int distance(int x, int y, vector<int>& target){
        return abs(x - target[0]) + abs(y - target[1]);
    }
    
public:
    int minDistance(int height, int width, vector<int>& tree, vector<int>& squirrel, vector<vector<int>>& nuts) {
        int totalDist = 0, squirrelToClosest = INT_MAX;
        
        for (vector<int>& nut : nuts){
            totalDist += 2 * distance(nut[0], nut[1], tree);
            // find the shortest distance between the a squirrel and nut with deducting a distance between the tree and nut.
            squirrelToClosest = min(squirrelToClosest, -distance(nut[0], nut[1], tree) + distance(nut[0], nut[1], squirrel));
        }
        
        return totalDist + squirrelToClosest;
    }
};